IPA-End-to-End.md

```diff
@@ -120,15 +120,28 @@ These events will occur on different _websites/apps_, so they then need to be ga
+* A _source fanout query_ is designed to help a _website/app_ understand the effect that the ads it shows have on outcomes for its advertisers.  A _source fanout query_ can only contain _source reports_ from a single _source website/app_.  A _source fanout query_ can include _trigger reports_ from multiple sites, where each report might represent a conversion event.
 
-When making a _fanout query_, the _report collector_ uses the information it receives about events to decide whether to include an event in the query and what value is attached to the event.  The _report collector_ also assigns a _breakdown key_ to each _source event_ and a _trigger value_ to each _trigger value_.  The _report collector_ also assigns a value of its choice for the _attribution constraint ID_ of all events.
+* A _trigger fanout query_ helps a site that buys ads to understand how its advertising is helping to drive outcomes on its _website/app_.  A _trigger fanout query_ can only contain _trigger reports_ from a single _source website/app_.  A _trigger fanout query_ includes _source reports_ from multiple sites, where each report might represent an ad impression or click.
+
+Before making a query, _source websites/apps_ and _trigger websites/apps_ respectively generate _source reports_ and _trigger_reports_ by associating _encrypted match keys_ with other values. A collection of _source reports_ and _trigger reports_ with associated values determine what a query result means:
+
+* A _breakdown key_ is associated with each _source report_ in a query.  The _breakdown key_ is used to split _source reports_ into groups for aggregation.  A _breakdown key_ can represent an advertising campaign, an advertising channel, a set of creatives, or any combination of these with other data from the event.  The output of the query includes an aggregate value for each _breakdown key_.
+
+* A _trigger value_ is associated with each _trigger report_.  If the _trigger report_ is matched with a _source reports_ by the MPC, the _trigger value_ contributes to the aggregated value for the _breakdown key_ from the _source report_.  Note that [sensitivity capping](#user-level-sensitivity-capping) might mean that some _trigger values_ do not contribute to the final result.
+
+* A _attribution constraint ID_ can be added to both _source reports_ and _trigger reports_.  Reports with different _attribution constraint IDs_ will not be matched to each other, which allows a site to include _source reports_ and _trigger reports_ that it does not wish to have attributed together.  For instance, if the goal is to perform attribution for distinct product segments, assigning a different _attribution constraint ID_ to reports from each distinct segment will ensure that impressions from one segment is not attributed to conversions from the other segment.
```

Comment:
```suggestion
* A _attribution constraint ID_ can be associated to both _source reports_ and _trigger reports_.  Reports with different _attribution constraint IDs_ will not be matched to each other, which allows a site to include _source reports_ and _trigger reports_ that it does not wish to have attributed together.  For instance, if the goal is to perform attribution for distinct product segments, assigning a different _attribution constraint ID_ to reports from each distinct segment will ensure that impressions from one segment is not attributed to conversions from the other segment.
```

Reply:
should we continue to use this verb?

Reply:
```suggestion
* A _attribution constraint ID_ can be associated with both _source reports_ and _trigger reports_.  Reports with different _attribution constraint IDs_ will not be matched to each other, which allows a site to include _source reports_ and _trigger reports_ that it does not wish to have attributed together.  For instance, if the goal is to perform attribution for distinct product segments, assigning a different _attribution constraint ID_ to reports from each distinct segment will ensure that impressions from one segment is not attributed to conversions from the other segment.
```

Reply:
ha yes, if we use the same verb we should use it correctly!